persistence/src/test/java/cz/muni/fi/pa165/mushrooms/dao/VisitDaoTest.java

```diff
@@ -151,9 +148,237 @@ public void create_validVisit() throws Exception {
+        Forest foudForest = em.find(Forest.class, forest1.getId());
+        Visit foundVisit = em.find(Visit.class, visit1.getId());
+
+        foudForest.setName(forestNewName);
+
+        foundVisit.setForest(foudForest);
+        visitDao.update(foundVisit);
+        em.flush();
+
+        assertThat(em.find(Visit.class, visit1.getId()).getForest().getName()).isEqualTo(forestNewName);
+    }
+
+    @Test
+    public void update_validVisitForestDescription() {
+        String forestNewDescription = "Dark and hard to find anything";
+        Forest foundForest = em.find(Forest.class, forest1.getId());
+        Visit foundVisit = em.find(Visit.class, visit1.getId());
+
+        foundForest.setDescription(forestNewDescription);
+
+        foundVisit.setForest(foundForest);
+        visitDao.update(foundVisit);
+        em.flush();
+
+        assertThat(em.find(Visit.class, visit1.getId()).getForest().getDescription()).isEqualTo(forestNewDescription);
+    }
+
+    @Test
+    public void update_validVisitHunterName() {
+        String hunterNewName = "Harry";
+        MushroomHunter foudHunter = em.find(MushroomHunter.class, mushroomHunter1.getId());
+        Visit foundVisit = em.find(Visit.class, visit1.getId());
+
+        foudHunter.setFirstName(hunterNewName);
+
+        foundVisit.setHunter(foudHunter);
+        visitDao.update(foundVisit);
+        em.flush();
+
+        assertThat(em.find(Visit.class, visit1.getId()).getHunter().getFirstName()).isEqualTo(hunterNewName);
+    }
+
+    @Test
+    public void update_validVisitHunterSurname() {
+        String hunterNewSurname = "Potter";
+        MushroomHunter foudHunter = em.find(MushroomHunter.class, mushroomHunter1.getId());
+        Visit foundVisit = em.find(Visit.class, visit1.getId());
+
+        foudHunter.setSurname(hunterNewSurname);
+
+        foundVisit.setHunter(foudHunter);
+        visitDao.update(foundVisit);
+        em.flush();
+
+        assertThat(em.find(Visit.class, visit1.getId()).getHunter().getSurname()).isEqualTo(hunterNewSurname);
+    }
+
+    @Test
+    public void update_validVisitHunterNickName() {
+        String hunterNickName = "The boy who survived";
+        MushroomHunter foudHunter = em.find(MushroomHunter.class, mushroomHunter1.getId());
+        Visit foundVisit = em.find(Visit.class, visit1.getId());
+
+        foudHunter.setUserNickname(hunterNickName);
+
+        foundVisit.setHunter(foudHunter);
+        visitDao.update(foundVisit);
+        em.flush();
+
+        assertThat(em.find(Visit.class, visit1.getId()).getHunter().getUserNickname()).isEqualTo(hunterNickName);
+    }
+
+
+    @Test
+    public void update_nullVisit() {
+        assertThatThrownBy(() ->visitDao.update(null)).hasRootCauseInstanceOf(IllegalArgumentException.class);
+    }
+
+    @Test
+    public void update_visitWithNullHunter() {
+        Forest testForest = new Forest();
+        testForest.setName("Test Forest");
+
+        em.persist(testForest);
+
```

Comment:
Isn't it redundant, if you have created forest1 and forest2 in @Before? Why don't you use one of them?

Reply:
thanks, changed as well